AI

Petzys opened the pull request.

Tayhm committed:
Merge remote-tracking branch 'origin/network_client' into ai

Tayhm committed:
fixed bugs

Tayhm committed:
fixed even more bugs

Tayhm committed:
Merge remote-tracking branch 'origin/network' into ai

Tayhm committed:
fixed bugs and also fixed AI chet messages

Tayhm committed:
[TTTK-15], [TTTK-54]: fixed a few things. now the AIs work

Tayhm committed:
Merge remote-tracking branch 'origin/network' into ai

Petzys committed:
Feat(ai): TTTK-21 TTTK-54 Fixed tests

JM-Lemmi commented on AI/ai_strategy.py:
```diff
@@ -113,12 +114,16 @@ def __init__(self, second_player: bool = False):
         self._ai_uuid = "108eaa05-2b0e-4e00-a190-8856edcd56a5"
         self._ai_uuid2 = "108eaa05-2b0e-4e00-a190-8856edcd56a6"
```
warum ist es denn die gleiche uuid 2 mal?

JM-Lemmi replied:
generate uuid2 randomly

```suggestion
        self._ai_uuid2 = "d90397a5-a255-4101-9864-694b43ce8a6c"
```

JM-Lemmi commented on AI/ai_strategy.py:
```diff
@@ -132,12 +137,16 @@ def __init__(self, second_player: bool = False):
         self._ai_uuid = "108eaa05-2b0e-4e00-a190-8856edcd56a5"
         self._ai_uuid2 = "108eaa05-2b0e-4e00-a190-8856edcd56a6"
```
generate uuid2 randomly

```suggestion
        self._ai_uuid2 = "d90397a5-a255-4101-9864-694b43ce8a6c"
```

HOOK-Hawkins committed:
Merge branch 'network' into ai

HOOK-Hawkins committed:
feat(AI): Add AI options to GUI controls [TTTK-21, TTTK-54]

HOOK-Hawkins committed:
fix(AI): Start AI thread as deamon to close on application close [TTTK-21, TTTK-54]

HOOK-Hawkins committed:
refactor(AI): refactor AI to avoid dublicate code, add different and compliant uuids
when spawning two ais of the same strength, the second needs to be initialised with a custom uuid

HOOK-Hawkins commented on Client/client.py:
```diff
@@ -170,7 +173,11 @@ async def _preprocess_message(self, message:str) -> str:
+                    self._opponent = self._lobby_status[0] if self._lobby_status[0] != self._player else self._lobby_status[1]
+
+                    if self._opponent == self._player:
+                        logger.info("ERROR WTF ALTER")
```
```suggestion
                        logger.error("player and opponent are equal")

```

JM-Lemmi committed:
fix(client): meaningful error message for equal opponents

Co-authored-by: Hauke Platte <53154723+HOOK-Hawkins@users.noreply.github.com>

JM-Lemmi committed:
Merge branch 'main' into ai
